dist/activate

```diff
@@ -0,0 +1,171 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# set up the environment variables
+KCTF_CONFIG_DIR="${HOME}/.config/kctf"
+KCTF_CTF_DIR="$(realpath --no-symlinks "$(dirname "${BASH_SOURCE-$0}")/..")"
+KCTF_CTF_NAME=$(basename "${KCTF_CTF_DIR}")
+KCTF_BIN="${KCTF_CTF_DIR}/kctf/bin"
+KCTF_SESSION="$(dd if=/dev/urandom bs=1 count=10 2>/dev/null | xxd -ps -c 10)"
+
+function _kctf_download_dependencies {
+  if [[ ! -x "${KCTF_BIN}/yq" ]]; then
+    if [[ -e "${KCTF_BIN}/yq" ]]; then
+      rm "${KCTF_BIN}/yq" >/dev/null
+    fi
+    wget "https://github.com/mikefarah/yq/releases/download/v4.2.0/yq_linux_amd64" -O "${KCTF_BIN}/yq" --quiet || return 1
+    sha256sum --status -c <(echo "5d44bd64e264e9029c5f06bcd960ba162d7ed7ddd1781f02a28d62f50577b632  ${KCTF_BIN}/yq") || return 1
+    chmod u+x "${KCTF_BIN}/yq"
+  fi
+}
+
+function _kctf_export_variables {
+  export KCTF_CTF_DIR
+  export KCTF_CTF_NAME
+  export KCTF_BIN
+  export CLOUDSDK_ACTIVE_CONFIG_NAME="kctf-${KCTF_SESSION}"
+  export KUBECONFIG="${KCTF_CONFIG_DIR}/kube-${KCTF_SESSION}.conf"
+}
+
+function _kctf_cleanup {
+  if command -v gcloud >/dev/null 2>&1; then
+    unset CLOUDSDK_ACTIVE_CONFIG_NAME
+    # regenerate this name in case the user changed the variable
+    GCLOUD_CONFIG_NAME="kctf-${KCTF_SESSION}"
+    if gcloud config configurations describe "${GCLOUD_CONFIG_NAME}" >/dev/null 2>&1; then
+      echo "Deleting gcloud config ${GCLOUD_CONFIG_NAME}"
+      CLOUDSDK_CORE_DISABLE_PROMPTS=1 gcloud config configurations delete "${GCLOUD_CONFIG_NAME}"
+    fi
+  fi
+  # regenerate this name in case the user changed the variable
+  KUBE_CONFIG_NAME="${KCTF_CONFIG_DIR}/kube-${KCTF_SESSION}.conf"
+  if [[ -e "${KUBE_CONFIG_NAME}" ]]; then
+    rm "${KUBE_CONFIG_NAME}" >/dev/null
+  fi
+}
+
+function _kctf_usage {
+  echo -e "usage: kctf command subcommand [args]" >&2
```

Comment:
nit:
```bash
cat >&2 <<EOF
Usage: kctf command subcommand [args]

Available commands:
  chal:      commands for managing challenges (create, start, stop, debug, ...)
  cluster:  commands for managing the cluster (create, start, stop, resize, ...)
  config:   commands for managing the kctf config (create, load)
EOF
```

Reply:
That way I have to left align everything to the beginning of the line don't I?

Reply:
yep, it's a styling thing - in python this is the usual way to do it (with `"""..."""`)

Reply:
In my opinion it looks pretty bad since it doesn't follow the indentation of the code block. I.e. I would use it only for unindented code